### PR TITLE
Properly set thread affinity for throughput tests

### LINKED ISSUE #917: User Mode Download Performance is Significantly Lower than Upload
### Describe the bug

For some reason, download perf is a lot worse than upload for user mode. We need to investigate why.

image

### PR DESCRIPTION
**Pull request by ThadHouse**
We were not setting this before, causing low performance for download performance tests. 

Should fix #917

### CODE REVIEW

**Comment by ThadHouse** [comment]
Doing this slightly differently. Realizing we shouldn't affinitize this to the core, but instead should just affinitize it to thread group 0.

File: src/perf/lib/ThroughputClient.cpp [hunk 1]
```diff
@@ -26,7 +26,6 @@ PrintHelp(
         "  -target:<####>              The target server to connect to.\n"
 #if _WIN32
         "  -comp:<####>                The compartment ID to run in.\n"
-        "  -core:<####>                The CPU core to use for the main thread.\n"
```
Comment (nibanks) [inline src/perf/lib/ThroughputClient.cpp:29]: I still think it'd be nice to support this flag for now.

**Review by nibanks — APPROVED** [review]

**Comment by nibanks** [comment]
Looks like Windows builds don't like `Affinity.Mask = (KAFFINITY)(~0ull);` Maybe try `Affinity.Mask = (KAFFINITY)(-1);` instead?

File: src/core/connection.c [hunk 2]
```diff
@@ -65,7 +65,7 @@ QuicConnAlloc(
     )
 {
     BOOLEAN IsServer = Datagram != NULL;
-    uint32_t CurProcIndex = QuicProcCurrentNumber();
+    uint32_t CurProcIndex = QuicProcCurrentNumber() & 0xFFFFFFFE;
```
Comment (nibanks) [inline src/core/connection.c]: No. We cannot do this.

Comment (ThadHouse) [reply src/core/connection.c]: I know, I was just trying it.

**Review by nibanks — APPROVED** [review]

**Pull request merged**